web/src/lib/server/google/gmail.ts

```diff
@@ -67,3 +67,130 @@ export const sendMessage = async (
 		body: JSON.stringify({ raw: urlSafeBase64(raw) })
 	});
 };
+
+/**
+ * Retrieves the IDs of all Gmail labels that start with "@SRC"
+ *
+ * @async
+ * @param {string} accessToken - A valid access token for the Gmail API
+ * @returns {Array<string>} - An array of the IDs of all Gmail labels that start with "@SRC"
+ * @throws Throws an error if the API request fails or if the response is invalid
+ */
+export const getSrcLabelIds = async (accessToken: string): Array<string> => {
+
+    // The Gmail labels endpoint, gives the full list of labels used in the users Gmail
+    const labelsEndpoint = `https://gmail.googleapis.com/gmail/v1/users/me/labels`;
```

Comment:
Nit: some of these variables in the functions are called `labelsEndpoint` `threadEndpoint` and others just `endpoint`. Makes sense to call of them `endpoint` since there is one per function.